### PR TITLE
Add open context to certain commands

### LINKED ISSUE #85636: TreeDataProvider: allow single click and double click to work the same way they do in the file exporer
I can add a command to activate when an item is selected, but I want the same behavior as the in-built explorer where single click opens the tab with the file (title in italics) and double click opens it permanently as a new tab. 

Can this be added to the TreeDataProvider interface?

### PR DESCRIPTION
**Pull request by bpasero**
This PR does 2 things:
* moves the commands `_workbench.open` and friends into core workbench (this helps to depend on the command IDs from core parts in the future)
* allows to pass `IOpenEvent` as argument to these commands in case invoked from trees

This PR fixes #85636

**PS:** I also made `EditorViewColumn` a core thing as well (renamed to `EditorGroupColumn`)

### CODE REVIEW

**Comment by bpasero** [comment]
https://github.com/microsoft/vscode/pull/110475/files#diff-ee44b7286aa52f5c5a2737955cebe3a73945354241d92bcfd8708cca313600b0R469-R473 is a bit of magic, but I did not want to overengineer it. If someone has a better idea, let me know...

**Review by jrieken — APPROVED** [review]
lgtm - I also think that the check for command ids in the tree is a little hacky, esp when we copy that for SCM and the timeline. But maybe we come up with something before this starts to spread

File: src/vs/workbench/contrib/views/browser/treeView.ts [hunk 1]
```diff
@@ -462,8 +463,16 @@ export class TreeView extends Disposable implements ITreeView {
 				return;
 			}
 			const selection = this.tree!.getSelection();
-			if ((selection.length === 1) && selection[0].command) {
-				this.commandService.executeCommand(selection[0].command.id, ...(selection[0].command.arguments || []));
+			const command = selection[0].command;
```
Comment (alexr00) [inline src/vs/workbench/contrib/views/browser/treeView.ts:466]: I think that `getSelection` can return an empty array.

**Review by alexr00 — APPROVED** [review]

File: src/vs/workbench/contrib/views/browser/treeView.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/contrib/views/browser/treeView.ts]: Thanks, fixed.

**Comment by bpasero** [comment]
@alexr00 please take a look at the JSDoc comment I adde to the `command` of `TreeItem` (https://github.com/microsoft/vscode/pull/110475/commits/8b0b052b120c294bc457ae898524fc82d89cb159)

> Please use `vscode.open` or `vscode.diff` as command IDs when the tree item is opening
> something in the editor. Using these commands ensures that the resulting editor will
> appear consistent with how other built-in trees open editors.

**Pull request merged**